src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java

```diff
@@ -173,32 +203,49 @@ public static PackageMetadataConfig createFromString(String yamlContents) {
 
   private static Map<TargetLanguage, VersionBound> createVersionMap(
       Map<String, Map<String, String>> inputMap) {
-    ImmutableMap.Builder<TargetLanguage, VersionBound> versionMap = new ImmutableMap.Builder<>();
-    for (Map.Entry<String, Map<String, String>> entry : inputMap.entrySet()) {
-      putWithDefault(
-          versionMap,
-          entry.getKey(),
-          VersionBound.create(entry.getValue().get("lower"), entry.getValue().get("upper")));
-    }
-    return versionMap.build();
+    Map<TargetLanguage, Map<String, String>> intermediate = buildMapWithDefault(inputMap);
+    // Convert parsed YAML map into VersionBound object
+    return Maps.transformValues(
```

Comment:
The previous logic was wrong; it put the default value as the key for every language instead of only for the unconfigured languages.